packages/playground/src/components/Playground.tsx

```diff
@@ -0,0 +1,366 @@
+                      }}
+                    />
+                  )}
+                </>
+              }
+              style={{
+                width: '100%',
+                height: 1000,
+                border: 0,
+              }}
+              theme={theme}
+            >
+              <FormComponent
+                {...templateProps}
+                {...liveSettings}
+                schema={schema}
+                uiSchema={uiSchema}
+                formData={formData}
+                onChange={onFormDataChange}
+                noHtml5Validate={true}
+                onSubmit={({ formData }: { formData: string }, e: any) => {
+                  console.log('submitted formData', formData);
+                  console.log('submit event', e);
+                  window.alert('Form submitted');
+                }}
+                fields={{ geo: GeoPosition }}
+                validator={validators[validator]}
+                onBlur={(id: string, value: string) => console.log(`Touched ${id} with value ${value}`)}
+                onFocus={(id: string, value: string) => console.log(`Focused ${id} with value ${value}`)}
+                onError={log('errors')}
+                ref={playGroundFormRef}
+              />
+            </DemoFrame>
+          )}
+        </ErrorBoundary>
+      </div>
+    </>
+  );
+};
+
+export default Playground;
+
+type EditorsProps = {
+  schema: unknown;
+  setSchema: React.Dispatch<React.SetStateAction<object>>;
+  uiSchema: unknown;
+  setUiSchema: React.Dispatch<React.SetStateAction<object>>;
+  formData: unknown;
+  setFormData: React.Dispatch<React.SetStateAction<object>>;
+  extraErrors: unknown;
+  setExtraErrors: React.Dispatch<React.SetStateAction<unknown>>;
+  setShareURL: React.Dispatch<React.SetStateAction<string | null>>;
+};
+
+const Editors: React.FC<EditorsProps> = ({
+  extraErrors,
+  formData,
+  schema,
+  uiSchema,
+  setExtraErrors,
+  setFormData,
+  setSchema,
+  setShareURL,
+  setUiSchema,
+}) => {
+  const onSchemaEdited = useCallback((newSchema) => {
+    setSchema(newSchema);
+    setShareURL(null);
+  }, []);
+
+  const onUISchemaEdited = useCallback((newUiSchema) => {
+    setUiSchema(newUiSchema);
+    setShareURL(null);
+  }, []);
+
+  const onFormDataEdited = useCallback((newFormData) => {
+    if (
+      !isEqualWith(newFormData, formData, (newValue, oldValue) => {
+        // Since this is coming from the editor which uses JSON.stringify to trim undefined values compare the values
+        // using JSON.stringify to see if the trimmed formData is the same as the untrimmed state
+        // Sometimes passing the trimmed value back into the Form causes the defaults to be improperly assigned
+        return JSON.stringify(oldValue) === JSON.stringify(newValue);
+      })
+    ) {
+      setFormData(newFormData);
+      setShareURL(null);
+    }
+  }, []);
```

Comment:
You probably need the dependencies for `formData`
```suggestion
  }, [formData]);
```